go/libraries/doltcore/merge/merge_prolly_rows.go

```diff
@@ -88,12 +88,12 @@ func mergeProllyTable(ctx context.Context, tm *TableMerger, mergedSch schema.Sch
 		// After we migrate the data on the left-side to the new, merged schema, we reset
 		// the left mapping to an identity mapping, since it's a direct mapping now.
 		valueMerger.leftMapping = val.NewIdentityOrdinalMapping(len(valueMerger.leftMapping))
-	}
-
-	// After we've migrated the existing data to the new schema, it's safe for us to update the schema on the table
-	mergeTbl, err = tm.leftTbl.UpdateSchema(sqlCtx, mergedSch)
-	if err != nil {
-		return nil, nil, err
+	} else {
```

Comment:
I don't think this should be in an else block... it seems like the first part of this function would migrate the table data, but then it wouldn't go into the else block to actually update the schema metadata of the table, so the schema and data would be out of sync.